src/input/keyboard/mod.rs

```diff
@@ -838,3 +1020,15 @@ impl<D: SeatHandler + 'static> KeyboardGrab<D> for DefaultGrab {
         unreachable!()
     }
 }
+
+mod private {
+    use super::xkb;
+    // Glue trait to avoid duplicated code.
+    pub trait XkbTypesHandle {
+        /// Get the reference to the xkb keymap.
+        fn keymap(&self) -> &xkb::Keymap;
+
+        /// Get the reference to the xkb state.
+        fn state(&self) -> &xkb::State;
+    }
```

Comment:
Trick to not expose `xkbcommon` crate.

Reply:
Isn't it already exposed?

> `pub use xkbcommon::xkb::{self,`

Reply:
Hm, but where you can build it then and how it's being used?

Reply:
The are used in `mods`, so I sort of exposed it.